jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/DelayedHandlerTest.java

```diff
@@ -179,6 +177,74 @@ public void testOnContent() throws Exception
             @Override
             public boolean process(Request request, Response response, Callback callback) throws Exception
             {
+                // Check that we are not called via any demand callback
+                ByteArrayOutputStream out = new ByteArrayOutputStream(8192);
+                new Throwable().printStackTrace(new PrintStream(out));
+                String stack = out.toString(StandardCharsets.ISO_8859_1);
+                assertThat(stack, not(containsString("DemandContentCallback.succeeded")));
+                assertThat(stack, not(containsString("UntilContentDelayedProcess.onContent")));
```

Comment:
These two assertions are quite weak as renaming the classes would render them meaningless.

Reply:
Yep.  But how else do you test for the negative?

Reply:
I made a bit less fragile by using reflection (there is a sentence I never thought I'd type :)